### PR TITLE
Consolidate mock fetch response helpers

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
While writing some unit tests for HCA redux actions (https://github.com/department-of-veterans-affairs/vets-website/pull/10564) I noticed we had multiple definitions of functions that mock fetch responses. This PR moves those functions into the common unit test helper file and uses them in place of the versions that were defined ad hoc across multiple test files.

## Testing done
All existing unit tests pass

## Screenshots
N/A

## Acceptance criteria
- [x] All existing unit tests pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/testing/unit/helpers.js [hunk 1]
```diff
@@ -99,6 +99,39 @@ export function mockFetch(returnVal, shouldResolve = true) {
     );
 }
 
+export function setFetchJSONResponse(stub, data = null) {
```
Comment (erikphansen) [inline src/platform/testing/unit/helpers.js]: This PR looks a little scarier than it really is. The bulk of the work is taking existing helper functions and moving them into this common unit test helpers file. And then using these new common helpers in the test files.

**Review by rianfowler — APPROVED** [review]

File: src/applications/personalization/connected-accounts/tests/actions/index.unit.spec.js [hunk 2]
```diff
@@ -3,28 +3,12 @@ import sinon from 'sinon';
 import {
   mockFetch,
   resetFetch,
-} from '../../../../../platform/testing/unit/helpers.js';
+  setFetchJSONFailure as setFetchFailure,
```
Comment (cvalarida) [inline src/applications/personalization/connected-accounts/tests/actions/index.unit.spec.js]: I'm seeing a lot of these renames. What do you think about naming the functions `setFetchFailure` and `setFetchResponse` if that's the name that makes the most sense in a test?

**Review by cvalarida — APPROVED** [review]
:shipit:

The dapper squirrel is pleased with this one.

File: src/applications/personalization/connected-accounts/tests/actions/index.unit.spec.js (hunk 2 above)
Comment (erikphansen) [reply src/applications/personalization/connected-accounts/tests/actions/index.unit.spec.js]: Well we have `setFetchJSONResponse` and `setFetchBlobResponse` defined next to each other in the helpers file. So I think having those different, specific names makes sense. I feel less strongly about if we should rename them on import in the consuming test files. I'm happy to kill these renames and use them with their "native" function names if you want.

**Pull request merged**